tests/api/testcl.hpp

```diff
@@ -755,5 +760,28 @@ class WithCommandQueue : public WithContext {
 
 class WithProfiledCommandQueue : public WithCommandQueue {
 protected:
-    void SetUp() override { SetUpQueue(CL_QUEUE_PROFILING_ENABLE); }
+    void SetUp() override {
+        WithContext::SetUp();
+        SetUpQueue(CL_QUEUE_PROFILING_ENABLE);
+    }
+};
+
+class WithPrintfEnabled : public WithCommandQueue {
+protected:
+    void SetUp() override{};
+    void TearDown() override {
+        cl_int err = clReleaseContext(m_context);
+        ASSERT_CL_SUCCESS(err);
+        if (m_queue != nullptr) {
+            WithCommandQueue::TearDown();
+        }
+        if (m_context != nullptr) {
+            WithContext::TearDown();
```

Comment:
It will have already been called by `WithCommandQueue::TearDown` if `m_queue != nullptr`.